tests/php/test-amp-style-sanitizer.php

```diff
@@ -2409,4 +2413,91 @@ public function test_prioritized_stylesheets( $html_generator, $assert ) {
+			// Twentytwenty inline styles.
+			[ AMP_DOM_Utils::create_node( $dom, 'style', [ 'id' => 'twentytwenty-style-inline-css' ] ), 2 ],
+			// Admin bar inline styles.
+			[ AMP_DOM_Utils::create_node( $dom, 'style', [ 'id' => 'admin-bar-inline-css' ] ), 200 ],
+			// Customizer inline styles.
+			[ AMP_DOM_Utils::create_node( $dom, 'style', [ 'id' => 'wp-custom-css' ] ), 60 ],
+			// Other inline styles.
+			[ AMP_DOM_Utils::create_node( $dom, 'style', [ 'id' => 'something-else' ] ), 70 ],
+			// Non-AMP handle for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'link', [ 'id' => 'mediaelement-css', 'media' => 'print' ] ), 1100 ],
+			// Admin bar handle for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'link', [ 'id' => 'admin-bar-css', 'media' => 'print' ] ), 300 ],
+			// Dashicons handle for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'link', [ 'id' => 'dashicons-css', 'media' => 'print' ] ), 190 ],
+			// Parent theme styles for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'link', [ 'href' => '//example.org/wp-content/themes/parent-theme/', 'media' => 'print' ] ), 101 ],
+			// Child theme styles for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'link', [ 'href' => '//example.org/wp-content/themes/child-theme/', 'media' => 'print' ] ), 110 ],
+			// Core frontend handle for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'link', [ 'id' => 'wp-block-library-css', 'media' => 'print' ] ), 120 ],
+			// Plugin asset for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'link', [ 'href' => '//example.org/wp-content/plugins/some-plugin', 'media' => 'print' ] ), 130 ],
+			// Other styles from wp-includes for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'link', [ 'href' => '//example.org/wp-includes/', 'media' => 'print' ] ), 140 ],
+			// All other links for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'link', [ 'id' => 'something-else', 'media' => 'print' ] ), 150 ],
+			// Twentytwenty inline styles for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'style', [ 'id' => 'twentytwenty-style-inline-css', 'media' => 'print' ] ), 102 ],
+			// Admin bar inline styles for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'style', [ 'id' => 'admin-bar-inline-css', 'media' => 'print' ] ), 300 ],
+			// Customizer inline styles for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'style', [ 'id' => 'wp-custom-css', 'media' => 'print' ] ), 160 ],
+			// Other inline styles for print.
+			[ AMP_DOM_Utils::create_node( $dom, 'style', [ 'id' => 'something-else', 'media' => 'print' ] ), 170 ],
+			// Style attribute.
+			[ $dom->createAttribute( 'something' ), 70 ],
+		];
+	}
+
+	/**
+	 * Test retrieval of stylesheet priorities.
+	 *
+	 * @covers       \AMP_Style_Sanitizer::get_stylesheet_priority()
+	 *
+	 * @dataProvider get_get_stylesheet_priority_data
+	 *
+	 * @param string $node     Node to check the priority of.
+	 * @param int    $expected Expected priority.
+	 */
+	public function test_get_stylesheet_priority( $node, $expected ) {
```

Comment:
The existing tests were not very thorough, so I added a more specific unit test for this method. This uncovered the bug with plugin CSS mentioned above.